plugins/vllm/superduper_vllm/model.py

```diff
@@ -1,171 +1,266 @@
 
-        sampling_params = SamplingParams(**kwargs)
-        results = self.engine.generate(prompts, sampling_params)
-        n = sampling_params.n
-        texts_outputs = []
-        for result in results:
-            text_outputs = [output.text for output in result.outputs]
-            if n == 1:
-                texts_outputs.append(text_outputs[0])
-            else:
-                texts_outputs.append(text_outputs)
-        return texts_outputs
-
-
-class VllmModel(BaseLLM):
-    """
-    Load a large language model from VLLM.
-
-    :param model_name: The name of the model to use.
-    :param tensor_parallel_size: The number of tensor parallelism.
-    :param trust_remote_code: Whether to trust remote code.
-    :param vllm_kwargs: Additional arguments to pass to the VLLM
-    :param on_ray: Whether to use Ray for parallelism.
-    :param ray_address: The address of the Ray cluster.
-    :param ray_config: The configuration for Ray.
+    :param vllm_params: Parameters for VLLM model
     """
 
-    model_name: str = dc.field(default="")
-    tensor_parallel_size: int = 1
-    trust_remote_code: bool = True
-    vllm_kwargs: dict = dc.field(default_factory=dict)
-    on_ray: bool = False
-    ray_address: t.Optional[str] = None
-    ray_config: dict = dc.field(default_factory=dict)
+    vllm_params: dict = dc.field(default_factory=dict)
 
     def __post_init__(self, db, artifacts):
-        self.on_ray = self.on_ray or bool(self.ray_address)
-        if "tensor_parallel_size" not in self.vllm_kwargs:
-            self.vllm_kwargs["tensor_parallel_size"] = self.tensor_parallel_size
+        super().__post_init__(db, artifacts)
+        assert "model" in self.vllm_params, "model is required in vllm_params"
+        self._async_llm = None
+        self._sync_llm = None
+        tensor_parallel_size = self.vllm_params.get("tensor_parallel_size", 1)
+        pipeline_parallel_size = self.vllm_params.get("pipeline_parallel_size", 1)
+        parallel_size = max(tensor_parallel_size, pipeline_parallel_size)
+        self.compute_kwargs["num_gpus"] = parallel_size
+
+    def _init_sync_llm(self):
+        if self._sync_llm is not None:
+            return
+        assert self._async_llm is None, "Cannot initialize both sync and async LLMs"
+        from vllm import LLM
 
-        if "trust_remote_code" not in self.vllm_kwargs:
-            self.vllm_kwargs["trust_remote_code"] = self.trust_remote_code
+        self._sync_llm = LLM(**self.vllm_params)
 
-        if "model" not in self.vllm_kwargs:
-            self.vllm_kwargs["model"] = self.model_name
+    async def _init_async_llm(self):
+        if self._async_llm is not None:
+            return
+        assert self._sync_llm is None, "Cannot initialize both sync and async LLMs"
+        from vllm import AsyncEngineArgs, AsyncLLMEngine
 
-        super().__post_init__(db, artifacts)
+        async_engine_args = AsyncEngineArgs(
+            # engine_use_ray=True,
```

Comment:
?

Reply:
This parameter has been deprecated in vllm, and I have already removed it.